pkg/blackboxexporter/blackboxexporter.go

```diff
@@ -0,0 +1,243 @@
+package blackboxexporter
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/openshift/route-monitor-operator/api/v1alpha1"
+	"github.com/openshift/route-monitor-operator/pkg/const/blackbox"
+	"github.com/openshift/route-monitor-operator/pkg/util/reconcile"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	//api's used
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	//local packages
+)
+
+type BlackboxExporter struct {
+	Client client.Client
```

Comment:
if you would have changed the code to look like
```
client.Client
```
you could just use the struct like:
```
b := BlackboxExporter{}
b.Client.List()
b.List() // this would also work
```

Reply:
Actually I like that the logic is separated because I can easily see which functions I use from the client, and  which belong to the struct itself. I use to think about inherenting other structs as a "is-a" relationship, vs a "has-a" relationship with a named member. Is one of the primary functions of this struct to act as a kubernetes client? No, so I'd separate it.

Reply:
oki, just wanted to use the shorthand.. can refactor to consolidate to your view.

I would like that this discussion to sit in a comment here / in a coding guide as this is a fine line I think we should keep note of when composing together interfaces / structs

Reply:
It's a matter of code style not functionality, so I don't see a reason to comment that each time I don't do it :D